pyscf/scf/hf.py

```diff
@@ -757,6 +787,60 @@ def level_shift(s, d, f, factor):
 def damping(f, f_prev, factor):
     return f*(1-factor) + f_prev*factor
 
+def make_sap(mol, sap_basis='sapgraspsmall'):
+    '''Superposition of atomic potentials (SAP) potential matrix
+
+    Args:
+        mol : MoleBase object
+            molecule for which SAP is computed
+        sap_basis : str
+            SAP basis name
+
+    Returns:
+        Vsap : ndarray
+            SAP potential matrix
+    '''
+    from pyscf.gto.basis import load, SAP_ALIAS
+    from pyscf.gto.mole import fakemol_for_cgtf_charge
+
+    atom_coords = numpy.asarray([coord[1] for coord in mol._atom])
+    atoms = numpy.asarray([coord[0] for coord in mol._atom])
+
+    sapbas = {
+        atom: numpy.asarray(
+                load((sap_basis), atom)[0][1:]
+            ) for atom in set(atoms)
+    }
+
+    logger.note(mol, f'Found SAP basis! Using {sap_basis.split("/")[-1]}')
```

Comment:
As I said in person, this routine should take in the SAP basis already as a dict. There are other potentials in the Basis Set Exchange, but this routine is limited to the internal basis set library in NWChem format.

Reply:
The load actually does look through the bse catalog if it doesn't find the basis locally.
But I see your point regarding the input argument sap_basis being a dict.

Reply:
But there does seem to be a slight annoyance with that: if pyscf doesn't find the data locally and ends up using the bse api, load returns a dict with one key-value pair: the selected atom symbol as key and the basis of that element as value. If the data is found locally load returns only the basis array without the element symbol as key. So the load function is kinda bad in that regard but oh well, I wrote a workaround.